src/helpers/options.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @description
+ * Merges multiple option objects into a base object while ignoring `undefined`
+ * values.
+ */
+export function mergeOptions<const T>(base: T, ...options: Partial<T>[]): T {
+  // We cannot use Object.assign because if the keys value is present and set
+  // to "undefined", it will overwrite the base value.
+  for (const option of options) {
+    for (const [key, value] of Object.entries(option)) {
+      if (value !== undefined) {
+        // We need this "as any" because Object.entries does not preserve the
+        // object or key types.
+        // TODO: We should fix this correctly by providing an Object.entries
+        // polyfill that preserves the types.
```

Comment:
It's fine, this is magic that operates outside of TS anyway. The signature for the helper method is enough typing.

```suggestion
```

Reply:
I added a TODO comment because it's enough for now, but in an ideal world, I'd have the correct typings all the way through without any hacks